### PR TITLE
feat: rewrite signup and signin page with tailwind UI

### LINKED ISSUE #100: [FEATURE] <Rewrite the signup/signin view with Tailwind UI>
### Description

We're most likely going to start using [Tailwind UI](https://github.com/ykdojo/defaang/discussions/98) for this project.

So we should rewrite the signup/singin view with maybe [this one](https://tailwindui.com/components/application-ui/forms/sign-in-forms#component-fef1d7e9849359af76d4d6ead796938b).

### PR DESCRIPTION
**Pull request by olamide203**
### Fixes
- a rewrite of the signup and signing page with tailwind UI

Closes #100 

### 📄 Note to reviewers

- used react-icons for social media icons on signup and sign in page

### 📷 Screenshots
Screenshot 2022-08-20 at 15-54-53 Defaang

Screenshot 2022-08-20 at 15-54-28 Defaang

Screenshot 2022-08-20 at 16-04-58 Defaang

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -1,2 +0,0 @@
-NEXT_PUBLIC_SUPABASE_URL=YOUR_SUPABASE_URL
-NEXT_PUBLIC_SUPABASE_ANON_KEY=YOUR_SUPABASE_ANON_KEY
```
Comment (ykdojo) [inline .env.template:2]: don't delete this file

File: pages/signin.tsx [hunk 2]
```diff
@@ -26,10 +26,39 @@ const SignIn: NextPage = () => {
...
-						<h1 className='p-2 text-3xl font-semibold'>Sign in ⚡</h1>
+					<div className='max-w-md p-4 lg:h-[500px] lg:w-[400px]'>
+						{/* <h1 className='p-2 text-3xl font-semibold'>Sign in ⚡</h1> */}
+						<div>
+							<h2 className='mt-6 text-start text-3xl font-bold tracking-tight text-gray-900'>
+								Sign in to your account
+							</h2>
+						</div>
+						<div className='mt-6'>
+							<p className='text-sm text-gray-700'>sign in with</p>
```
Comment (ykdojo) [inline pages/signin.tsx:39]: sign in -> Sign in

File: pages/signin.tsx [hunk 3]
```diff
@@ -38,37 +67,79 @@ const SignIn: NextPage = () => {
...
+								<div className='flex items-center'>
+									<input
+										id='remember-me'
+										name='remember-me'
+										type='checkbox'
+										className='h-4 w-4 rounded border-gray-300 text-indigo-600 focus:ring-indigo-500'
+									/>
+									<label htmlFor='remember-me' className='ml-2 block text-sm text-gray-900'>
+										{' '}
+										Remember me{' '}
```
Comment (ykdojo) [inline pages/signin.tsx:108]: do we have these functionalities yet?

Comment (ykdojo) [reply pages/signin.tsx:108]: (if not, let's comment them out or delete them)

File: pages/signup.tsx [hunk 4]
```diff
@@ -39,59 +65,71 @@ const SignUp: NextPage = () => {
...
 									placeholder='*****'
 									value={password}
 									onChange={e => setPassowrd(e.target.value)}
 								/>
 							</div>
-							<div className='flex w-[90%] flex-col space-y-2 '>
-								<p>Confirm Password</p>
+							<div className='flex w-full flex-col space-y-2 '>
+								<label htmlFor='confirm-password' className='text-sm text-gray-600'>
+									Confirm Password
```
Comment (ykdojo) [inline pages/signup.tsx:112]: what is this for?

File: pages/signup.tsx [hunk 5]
```diff
@@ -26,11 +26,37 @@ const SignUp: NextPage = () => {
...
 					{/* Now you can decide different login ways - primarily I am giving google and twiiter for more add it yourself. */}
-					<div className='h-full w-full  bg-white p-4 md:shadow-lg lg:h-[650px] lg:w-[450px]'>
-						<h1 className='p-2 text-3xl font-semibold'>Create an Account ⚡</h1>
+					<div className='h-full max-w-md p-4 lg:h-[650px] lg:w-[450px]'>
+						{/* <h1 className='p-2 text-3xl font-semibold'>Create an Account ⚡</h1> */}
+						<div>
+							<h2 className='mt-6 text-start text-3xl font-bold tracking-tight text-gray-900'>Create an Account</h2>
+						</div>
+						<div className='mt-6'>
+							<p className='text-sm text-gray-700'>sign up with</p>
```
Comment (ykdojo) [inline pages/signup.tsx:38]: sign up -> Sign up

File: pages/signup.tsx [hunk 6]
```diff
@@ -26,11 +26,37 @@ const SignUp: NextPage = () => {
...
-				<div className='flex h-screen  w-[100%] flex-col items-center justify-center bg-gray-100 lg:w-[50%]'>
+			<main className='flex h-max w-screen'>
+				<div className='flex w-[100%] flex-col items-center justify-center bg-gray-50 lg:w-[40%]'>
 					{/* Now you can decide different login ways - primarily I am giving google and twiiter for more add it yourself. */}
-					<div className='h-full w-full  bg-white p-4 md:shadow-lg lg:h-[650px] lg:w-[450px]'>
-						<h1 className='p-2 text-3xl font-semibold'>Create an Account ⚡</h1>
+					<div className='h-full max-w-md p-4 lg:h-[650px] lg:w-[450px]'>
+						{/* <h1 className='p-2 text-3xl font-semibold'>Create an Account ⚡</h1> */}
+						<div>
+							<h2 className='mt-6 text-start text-3xl font-bold tracking-tight text-gray-900'>Create an Account</h2>
```
Comment (ykdojo) [inline pages/signup.tsx:35]: Create an Account -> Create an account

File: pages/signup.tsx [hunk 7]
```diff
@@ -26,11 +26,37 @@ const SignUp: NextPage = () => {
 			<Head>
 				<title>Defaang / Welcome</title>
 			</Head>
-			<main className='flex h-screen w-screen'>
-				<div className='flex h-screen  w-[100%] flex-col items-center justify-center bg-gray-100 lg:w-[50%]'>
+			<main className='flex h-max w-screen'>
+				<div className='flex w-[100%] flex-col items-center justify-center bg-gray-50 lg:w-[40%]'>
 					{/* Now you can decide different login ways - primarily I am giving google and twiiter for more add it yourself. */}
```
Comment (ykdojo) [inline pages/signup.tsx:31]: delete this line

File: pages/signup.tsx [hunk 8]
```diff
@@ -26,11 +26,37 @@ const SignUp: NextPage = () => {
...
-					<div className='h-full w-full  bg-white p-4 md:shadow-lg lg:h-[650px] lg:w-[450px]'>
-						<h1 className='p-2 text-3xl font-semibold'>Create an Account ⚡</h1>
+					<div className='h-full max-w-md p-4 lg:h-[650px] lg:w-[450px]'>
+						{/* <h1 className='p-2 text-3xl font-semibold'>Create an Account ⚡</h1> */}
```
Comment (ykdojo) [inline pages/signup.tsx:33]: delete this line

File: pages/signin.tsx [hunk 9]
```diff
@@ -38,37 +67,79 @@ const SignIn: NextPage = () => {
...
-							onClick={handleSignIn}
-						>
-							Login
-						</button>
-						<p className='font-semibold'>
+
+						<p className='my-2 flex w-full justify-center text-sm text-gray-700'>
 							Not registered yet?
 							<Link className='cursor-pointer font-bold' href={'/signup'}>
 								<a className=' text-blue-600 hover:underline'> Sign up</a>
```
Comment (ykdojo) [inline pages/signin.tsx:145]: There should be a space before "Sign up"

**Review by ykdojo — CHANGES_REQUESTED** [review]
Thank you! It looks much better than what we had before, but I think the registration/signin box should be in a white background.

File: .env.template (hunk 1 above)
Comment (olamide203) [reply .env.template:2]: sorry, I renamed the original without making a copy first. would make a PR with changes

File: pages/signin.tsx (hunk 3 above)
Comment (olamide203) [reply pages/signin.tsx:108]: I don't think that functionality has been implemented yet. i'll comment out that part

File: pages/signup.tsx (hunk 4 above)
Comment (olamide203) [reply pages/signup.tsx:112]: just to confirm password before signup. it was in the former signup form so I added it to the new one also

Comment (olamide203) [reply pages/signup.tsx:112]: should I remove it?

**Comment by olamide203** [comment]
I've made changes to the areas that were pointed out
also used the image from the code template in place of the former one

###Screenshot
Screenshot 2022-08-20 at 19-08-42 Defaang

**Comment by ykdojo** [comment]
Thank you. The new UI has a lot of whitespace on the bottom, though:

image

Also, the "sign in with" options should be Google and GitHub - or we can comment them out for now.

File: pages/signup.tsx [hunk 10]
```diff
@@ -26,84 +25,159 @@ const SignUp: NextPage = () => {
...
-						>
-							Sign up
-						</button>
-						<p className='font-semibold'>
-							Already have an account?
+						<p className='my-2 flex w-full justify-center text-sm text-gray-700'>
+							Already have an account?{' '}
 							<Link className='cursor-pointer font-bold' href={'/signin'}>
-								<a className=' text-blue-600 hover:underline'> Sign in</a>
+								<a className=' mx-1 text-blue-600 hover:underline'> Sign in</a>
```
Comment (rohitdasu) [inline pages/signup.tsx:166]: please add a space before Sign in

File: pages/signin.tsx [hunk 11]
```diff
@@ -23,13 +22,74 @@ const SignIn: NextPage = () => {
...
+			<main className='flex min-h-full'>
 				{/* Left section - main login  */}
 
-				<div className='flex h-screen  w-[100%] flex-col items-center justify-center bg-gray-100 lg:w-[50%]'>
+				<div className='flex flex-1 flex-col justify-center py-12 px-4 sm:px-6 lg:flex-none lg:px-20 xl:px-24'>
 					{/* Now you can decide different login ways - primarily I am giving google and twiiter for more add it yourself. */}
-					<div className='w-full  bg-white p-4 md:shadow-lg lg:h-[500px] lg:w-[400px]'>
-						<h1 className='p-2 text-3xl font-semibold'>Sign in ⚡</h1>
+					<div className='mx-auto w-full max-w-sm lg:w-96'>
+						{/* <h1 className='p-2 text-3xl font-semibold'>Sign in ⚡</h1> */}
```
Comment (rohitdasu) [inline pages/signin.tsx:31]: please remove it if it's not needed

File: pages/signin.tsx [hunk 12]
```diff
@@ -23,13 +22,74 @@ const SignIn: NextPage = () => {
...
+								</div>
+
+								<div>
+									<a
+										href='#'
+										className='inline-flex w-full justify-center rounded-md border border-gray-300 bg-white py-2 px-4 text-sm font-medium text-gray-500 shadow-sm hover:bg-gray-50'
+									>
+										<span className='sr-only'>Sign in with Twitter</span>
+										<svg className='h-5 w-5' aria-hidden='true' fill='currentColor' viewBox='0 0 20 20'>
+											<path d='M6.29 18.251c7.547 0 11.675-6.253 11.675-11.675 0-.178 0-.355-.012-.53A8.348 8.348 0 0020 3.92a8.19 8.19 0 01-2.357.646 4.118 4.118 0 001.804-2.27 8.224 8.224 0 01-2.605.996 4.107 4.107 0 00-6.993 3.743 11.65 11.65 0 01-8.457-4.287 4.106 4.106 0 001.27 5.477A4.073 4.073 0 01.8 7.713v.052a4.105 4.105 0 003.292 4.022 4.095 4.095 0 01-1.853.07 4.108 4.108 0 003.834 2.85A8.233 8.233 0 010 16.407a11.616 11.616 0 006.29 1.84' />
```
Comment (rohitdasu) [inline pages/signin.tsx:63]: let's not directly put the svg code into the tsx.

File: pages/signin.tsx [hunk 13]
```diff
@@ -23,13 +22,74 @@ const SignIn: NextPage = () => {
...
+								<div>
+									<a
+										href='#'
+										className='inline-flex w-full justify-center rounded-md border border-gray-300 bg-white py-2 px-4 text-sm font-medium text-gray-500 shadow-sm hover:bg-gray-50'
+									>
+										<span className='sr-only'>Sign in with GitHub</span>
+										<svg className='h-5 w-5' aria-hidden='true' fill='currentColor' viewBox='0 0 20 20'>
+											<path
+												fillRule='evenodd'
+												d='M10 0C4.477 0 0 4.484 0 10.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0110 4.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.203 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.942.359.31.678.921.678 1.856 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0020 10.017C20 4.484 15.522 0 10 0z'
```
Comment (rohitdasu) [inline pages/signin.tsx:77]: let's not use svg code inside tsx

**Review by rohitdasu — CHANGES_REQUESTED** [review]

**Comment by olamide203** [comment]
### Fixes 

- Removed extra space at the bottom of  UI
- Used react-icons for social icons
- Replaced social signup options with google and github

File: package.json [hunk 14]
```diff
@@ -14,7 +14,8 @@
 		"@supabase/supabase-js": "^1.35.6",
 		"next": "12.2.5",
 		"react": "18.2.0",
-		"react-dom": "18.2.0"
+		"react-dom": "18.2.0",
+		"react-icons": "^4.4.0"
```
Comment (rohitdasu) [inline package.json:18]: @ykdojo @iShibi @subhoghoshX 
are you guys fine with `react-icons` package? IMO i could have downloaded svgs(fontawesome) and place it inside public folder and use it instead of adding a package.

Comment (ykdojo) [reply package.json:18]: I don't have a strong opinion on it as long as the performance is reasonably fast!

**Comment by ykdojo** [comment]
Thank you so much! It looks great.

A few minor things:

1. I feel like the heading ("Sign in to your account") is slightly too thick?

That might be a styling thing - I might make a separate issue for it instead:

image

2. Should we comment out the social logins/signups until we implement them?

**Comment by olamide203** [comment]
- The heading does look a bit wordy. I think just 'Sign in' should be okay.
- we could also comment out the social logins/signups till they are implemented. I think there is an open issue addressing it.
should I make a new PR with these changes?

**Comment by ykdojo** [comment]
Yes! Thank you. I think you can just update this PR

**Comment by ykdojo** [comment]
One more thing - let's remove the username field from the signup form.
image

Context: #95

**Comment by olamide203** [comment]
what about the confirm password field? it not currently in use. should it be commented out till the feature gets implemented?

**Comment by ykdojo** [comment]
Let's actually remove that one. I don't see a lot of websites using that field nowadays, and I don't think it's necessary.

File: package.json (hunk 14 above)
Comment (iShibi) [reply package.json:18]: Their docs state that tree-shaking works with this library. If that's true then it is fine to keep it as a dependency.

File: pages/signin.tsx [hunk 15]
```diff
@@ -38,49 +72,94 @@ const SignIn: NextPage = () => {
...
-							Login
-						</button>
-						<p className='font-semibold'>
-							Not registered yet?
+
+						<p className='my-2 flex w-full justify-center text-sm text-gray-700'>
+							Not registered yet?{' '}
 							<Link className='cursor-pointer font-bold' href={'/signup'}>
-								<a className=' text-blue-600 hover:underline'> Sign up</a>
+								<a className='mx-1 text-blue-600 hover:underline'> Sign up</a>
```
Comment (iShibi) [inline pages/signin.tsx:151]: No need to add space here as you already added a `{' '}` on line 149 after `Not registered yet?`
```suggestion
								<a className='mx-1 text-blue-600 hover:underline'>Sign up</a>
```

File: pages/signup.tsx [hunk 16]
```diff
@@ -1,109 +1,124 @@
...
 							className='mt-4 flex flex-col space-y-4'
 						>
-							<div className='flex w-[90%] flex-col space-y-2'>
-								<p>Email</p>
+							<div className='flex w-full flex-col space-y-2'>
+								<label htmlFor='email-address' className='text-sm text-gray-600'>
+									Email address
+								</label>
 								<input
 									type='text'
```
Comment (iShibi) [inline pages/signup.tsx:76]:
```suggestion
									type='email'
```

File: pages/signin.tsx [hunk 17]
```diff
@@ -38,49 +72,94 @@ const SignIn: NextPage = () => {
...
 							}}
 						>
-							<div className='flex w-[90%] flex-col space-y-2'>
-								<p>Email</p>
+							<div className='flex w-full flex-col space-y-3'>
+								<label htmlFor='email-address' className='text-sm text-gray-600'>
+									Email address
+								</label>
 								<input
 									type='text'
```
Comment (iShibi) [inline pages/signin.tsx:80]:
```suggestion
									type='email'
```

**Review by iShibi — CHANGES_REQUESTED** [review]

File: pages/signup.tsx (hunk 4 above)
Comment (iShibi) [reply pages/signup.tsx]: I suggest keeping the `Confirm Password` input as it forces users to double-check their password before signing up. Can you add this back?

Comment (ykdojo) [reply pages/signup.tsx]: @iShibi is there any modern website that uses that field? I just haven't really seen it recently.

Comment (ykdojo) [reply pages/signup.tsx]: Plus as long as the email is correct, the user can always reset their password. So if anything, there should be a "confirm email" field, but I don't think that's necessary, either.

Comment (iShibi) [reply pages/signup.tsx]: That's a valid point. I guess we are good without it.

**Comment by iShibi** [comment]
Looks good to me. We can use some `SignIn` and `SignUp` components here to keep the pages tidy, what do you think?

**Comment by olamide203** [comment]
yea, both forms could be moved to separate components with a `handleSubmit` prop which would get called on submit

**Comment by iShibi** [comment]
I think this is ready to merge. In the next iteration, we can add options for usernames and socials. Just run `npm run format:fix` to resolve the prettier error and we are good to go.

File: components/SignUp/Form.tsx [hunk 18]
```diff
@@ -0,0 +1,61 @@
+import { useState } from 'react';
+import { supabase } from '../../utils/supabaseClient';
+
+export const SignUpForm = () => {
+	const [email, setEmail] = useState('');
+	const [password, setPassowrd] = useState('');
```
Comment (benny123tw) [inline components/SignUp/Form.tsx:6]: typo `setPassowrd` -> `setPassword`

File: components/SignUp/Form.tsx [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+	const handleSignup = async () => {
```
Comment (benny123tw) [inline components/SignUp/Form.tsx:7]: maybe using camel case? `handleSignup` -> `handleSignUp`

**Comment by ykdojo** [comment]
Since I have to interact with this part of the code as I work on #108, I'm going to merge it as is.

I might raise issues/propose fixes as I go.
